### PR TITLE
feat: startup probe

### LINKED ISSUE #79: Bug: Startupprobe causes gap in data collection
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

If a sparrow pod is restarted, it causes a gap in monitoring. This is due to the healthcheck returning an HTTP 200 to the kubernetes probe. Since Kubernetes thinks the new sparrow pod is ready to accept traffic, prometheus scrape requests are sent to that new pod immediately, before the startup routine is actually finished. 

Screenshot from prometheus below

### Expected Behavior

A restart of the sparrow deployment waits for the new pod to have finished its first reconciliation cycle. Only then will the pod accept traffic. This ensures, that prometheus never scrapes a pod while still in startup.

### Steps To Reproduce

1. Spin up sparrow on kubernetes
2. Restart the deployment
3. HTTP GET /metrics or /openapi
You will notice sparrow returns no metrics for a few seconds after startup. It also does not show any checks in the /openapi endpoint 

### Relevant logs and/or screenshots, environment information, etc.

image

### Who can address the issue?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by lvlcn-t**
## Motivation

To implement a probe that checks if the pod is ready.

To be merged after #115 

Closes #79 

## Changes

Until #115 is merged, please look at this diff: https://github.com/caas-team/sparrow/compare/feat/healthz-endpoint...feat/startup-probe

I've added a startup and readiness probe to check if the pod is ready.

For additional information look at the commits.

## Tests done

Successfully deployed on local kind cluster:

Startup process:
image

image

## TODO

- [x] I've assigned this PR to myself
- [x] I've labeled this PR correctly

### CODE REVIEW

**Review by puffitos — COMMENTED** [review]
You've built out a nice PR, but you have you really addressed the problem at hand? When is the **sparrow** realy healthy? 

If the answer is, when the checks are running, then add a running field in each check's implementation and set it to true when you start the check. Iterate over all checks and see if all are running -> healthy. If a check is registered and isn't running, then something is wrong (is this even possible?)

Doing an HTTP request from the sparrow to the sparrow' checks' /metrics endpoint and checking whether the response is 200, querying the sparrow's own /metrics endpoint and checking for 200 all seems like overengineering. An HTTP Server, which send HTTP calls to itself to check whether it's running?

 To implement all this an interface was create and added to the already heavy sparrow struct and also the `api` OKhandler was exposed. 
 
 I think those are one too many changes, to just answer a simple question: is the sparrow healthy? The sparrow is healthy, when the OK-handler responds with an OK. Let's start with that; if we see, that more complex logic is needed for k8s, then we can implement it. 
 
 What you've tried to implement here is more like a /ready endpoint (the sparrow is ready to receive requests at /api /metrics /v1/metrics/checkName). Let's clear up first what the readiness and healthiness mean, and then implement it. 
 
 The checks and the sparrow are *separate*. We shouldn't forget our founding principles for the sparrow.

File: chart/templates/deployment.yaml [hunk 1]
```diff
@@ -75,6 +75,18 @@ spec:
           volumeMounts:
             - name: sparrow-config
               mountPath: /config/
+          readinessProbe:
```
[the remark this answers is not in the document]
Comment (puffitos) [reply chart/templates/deployment.yaml:78]: My comment above refers ofc to #115, I just went over the complete change and made the comment here.

[the remark this answers is not in the document]
Comment (lvlcn-t) [reply chart/templates/deployment.yaml:78]: I'll address it once I've merged #115. 

Also let's discuss your concern about my `/healthz` endpoint there.

[the remark this answers is not in the document]
Comment (lvlcn-t) [reply chart/templates/deployment.yaml:79]: Nvm, let's do it regardless of the discussion 😄

**Review by y-eight — APPROVED** [review]

**Pull request merged**